packages/application-config/test/process-config.spec.js

```diff
@@ -118,6 +118,41 @@ describe('processing a simple config', () => {
       });
     });
   });
+  describe('with MC_APP_ENV=staging and NODE_ENV=production', () => {
+    it('should process the config in production mode', () => {
+      const result = processConfig(
+        createTestOptions({
+          processEnv: {
+            NODE_ENV: 'production',
+            MC_APP_ENV: 'staging',
+          },
+        })
+      );
+      expect(result).toEqual({
+        env: {
+          applicationId: undefined,
+          applicationName: 'avengers-app',
+          cdnUrl: 'https://avengers.app/',
+          env: 'staging',
```

Comment:
Yeah basically we derive the "production" value from the env variables but also use it for this `env` field.

90% of the cases this is always production. Only for our internal apps it's nice to differentiate it with staging.
Because of that, I didn't want to accept an additional value only for this use case.